### PR TITLE
Ecp omp hackathon comp prim rel hydro

### PR DESCRIPTION
**Pull request by dunhamsj**
- Put auxiliary fluid fields on device at field creation level (FluidFieldsModule.F90)
- Modified ComputePrimitive subroutines for (non-)relativistic hydro to operate on 1-dimensional vectors
- Modified Euler_dgDiscretizationModule to be compatible with using 1D vectors
- Small re-write of TimersModule_Euler for aesthetics and increased ease of adding new timers

### CODE REVIEW

**Comment by jaharris87** [comment]
Before this proceeds, you'll need to resolve the conflicts, but don't do it in the browser.

Start by merging master into your branch if you haven't already.

**Comment by jaharris87** [comment]
Before this proceeds, we'll ned to resolve the conflicts, but not in the browser.

File: Modules/Euler/Euler_UtilitiesModule_NonRelativistic.F90 [hunk 1]
```diff
@@ -90,9 +91,9 @@ SUBROUTINE ComputePrimitive_Scalar &
     ( N, S_1, S_2, S_3, G, Ne, D, V_1, V_2, V_3, E, De, &
       Gm_dd_11, Gm_dd_22, Gm_dd_33 )
 
-#if defined(THORNADO_OMP_OL)
+#if   defined( THORNADO_OMP_OL ) && !defined( THORNADO_EULER_NOGPU )
```
Comment (jaharris87) [inline Modules/Euler/Euler_UtilitiesModule_NonRelativistic.F90:94]: We still need these routines on the GPU outside of just the Hydro runs, so don't add the `THORNADO_EULER_NOGPU` in here.

Comment (dunhamsj) [reply Modules/Euler/Euler_UtilitiesModule_NonRelativistic.F90:94]: Okay, I can remove those. I'll look around and see if that happened elsewhere

**Review by jaharris87 — APPROVED** [review]

**Pull request merged**